src/Controller/ForgotPasswordController.php

```diff
@@ -100,44 +103,37 @@ protected function getEmailTemplate(string $emailAddress, ResetPasswordToken $re
     public function checkEmail(SessionInterface $session): Response
     {
         // We prevent users from directly accessing this page
-        if (!$session->get(self::SESSION_CAN_CHECK_EMAIL)) {
+        if (!$this->canCheckEmail()) {
             return $this->redirectToRoute('app_forgot_password_request');
         }
 
-        $session->remove(self::SESSION_CAN_CHECK_EMAIL);
-
         return $this->render('forgot_password/check_email.html.twig', [
-            'tokenLifetime' => self::LIFETIME_HOURS,
+            'tokenLifetime' => $this->helper->getTokenLifetime(),
         ]);
     }
 
     /**
-     * @Route("/reset/{tokenAndSelector}", name="app_reset_password")
+     * @Route("/reset/{token}", name="app_reset_password")
      */
-    public function reset(Request $request, ResetPasswordHelperInterface $helper, UserPasswordEncoderInterface $passwordEncoder, string $tokenAndSelector = null): Response
+    public function reset(Request $request, UserPasswordEncoderInterface $passwordEncoder, string $token = null): Response
     {
         //Put token in session and redirect to self
-        if ($tokenAndSelector) {
+        if ($token) {
             // We store token in session and remove it from the URL,
             // to avoid any leak if someone get to know the URL (AJAX requests, Analytics...).
-            $request->getSession()->set(self::SESSION_TOKEN_KEY, $tokenAndSelector);
+            $this->storeTokenInSession($token);
 
             return $this->redirectToRoute('app_reset_password');
         }
 
         //Get token out of session storage
-        $tokenAndSelector = $request->getSession()->get(self::SESSION_TOKEN_KEY);
-        if (!$tokenAndSelector) {
+        $token = $this->getTokenFromSession();
+        if (!$token) {
             throw $this->createNotFoundException();
         }
 
         //Validate token using password helper
```

Comment:
probably not needed

Reply:
I have a new bug ticket in the bundle for the controller trait `getTokenFromSession()` . That method should return `?string`, currently it only returns `string`. Anyhow, `validateTokenAndFetchUser()` only accepts `string`.  I'm thinking that this conditional / exception should be added to the controller trait method.